From eb42dca2982f6f3461573667cb7f9bf8961f8269 Mon Sep 17 00:00:00 2001
From: Han Xu <han.xu@nxp.com>
Date: Fri, 2 Jun 2017 16:52:56 -0500
Subject: [PATCH 2015/5242] MLK-15284-2: dma: mxs-dma: add i.MX8QXP support in
 mxs-dma

commit  87105a8356699b1bb8ae284cc74ce31ce566eadc from
https://source.codeaurora.org/external/imx/linux-imx.git

add one more entry for i.MX8QXP mxs-dma, also check mxs_dma in filter
function.

Signed-off-by: Han Xu <han.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/dma/mxs-dma.c |   11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/dma/mxs-dma.c b/drivers/dma/mxs-dma.c
index 0704ded..93aa7b2 100644
--- a/drivers/dma/mxs-dma.c
+++ b/drivers/dma/mxs-dma.c
@@ -131,6 +131,7 @@ enum mxs_dma_id {
 	IMX23_DMA,
 	IMX28_DMA,
 	IMX7D_DMA,
+	IMX8QXP_DMA,
 };
 
 struct mxs_dma_engine {
@@ -167,6 +168,9 @@ struct mxs_dma_type {
 	}, {
 		.id = IMX7D_DMA,
 		.type = MXS_DMA_APBH,
+	}, {
+		.id = IMX8QXP_DMA,
+		.type = MXS_DMA_APBH,
 	}
 };
 
@@ -187,6 +191,9 @@ struct mxs_dma_type {
 		.name = "imx7d-dma-apbh",
 		.driver_data = (kernel_ulong_t) &mxs_dma_types[4],
 	}, {
+		.name = "imx8qxp-dma-apbh",
+		.driver_data = (kernel_ulong_t) &mxs_dma_types[5],
+	}, {
 		/* end of list */
 	}
 };
@@ -197,6 +204,7 @@ struct mxs_dma_type {
 	{ .compatible = "fsl,imx28-dma-apbh", .data = &mxs_dma_ids[2], },
 	{ .compatible = "fsl,imx28-dma-apbx", .data = &mxs_dma_ids[3], },
 	{ .compatible = "fsl,imx7d-dma-apbh", .data = &mxs_dma_ids[4], },
+	{ .compatible = "fsl,imx8qxp-dma-apbh", .data = &mxs_dma_ids[5], },
 	{ /* sentinel */ }
 };
 MODULE_DEVICE_TABLE(of, mxs_dma_dt_ids);
@@ -755,6 +763,9 @@ static bool mxs_dma_filter_fn(struct dma_chan *chan, void *fn_param)
 	struct mxs_dma_engine *mxs_dma = mxs_chan->mxs_dma;
 	int chan_irq;
 
+	if (!mxs_dma)
+		return false;
+
 	if (mxs_dma->dma_device.dev->of_node != param->of_node)
 		return false;
 
-- 
1.7.9.5

